bvm/ballerina-core/src/main/java/org/ballerinalang/util/metrics/Counter.java

```diff
@@ -91,23 +96,44 @@ public Counter register(MetricRegistry registry) {
         }
     }
 
+    /**
+     * Register the Metric to the registry.
+     */
+    default Counter register() {
+        return DefaultMetricRegistry.getInstance().register(this);
+    }
+
+    /**
+     * Unregisters the metric to the registry.
+     */
+    default void unregister() {
+        DefaultMetricRegistry.getInstance().unregister(this);
+    }
     /**
      * Increment the counter by one.
      */
     default void increment() {
-        increment(1D);
+        increment(1L);
     }
 
+    /**
+     * Reset to the counter's current value to zero.
+     *
+     */
+    void reset();
```

Comment:
Counters usually don't have a reset() method.

Reply:
This will help users to reuse the counter metrics without creating a new one if they want to reset.